src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Status.tsx

```diff
@@ -70,37 +69,40 @@ export const Status: React.FC<StatusProps> = ({ accountId }) => {
     selectMarketDataByAssetIdUserCurrency(state, assetId ?? ''),
   )
 
-  const account = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
-
-  const serializedTxIndex = useMemo(() => {
-    if (!(state?.txid && accountId && account)) return ''
-    return serializeTxIndex(accountId, state.txid, account)
-  }, [state?.txid, accountId, account])
-
-  const confirmedTransaction = useAppSelector(gs => selectTxById(gs, serializedTxIndex))
-
   useEffect(() => {
-    if (!accountId) return
+    if (!(contextDispatch && state?.txid?.length && accountId)) return
+    ;(async () => {
+      // Skipping outbound detection since there's no outbound tx involved here - as long as the inner swap is confirmed, we're gucci
+      const thorchainTxStatus = await waitForThorchainUpdate({
+        txId: state.txid!,
+        skipOutbound: true,
+      }).promise
 
-    if (confirmedTransaction && confirmedTransaction.status !== 'Pending' && contextDispatch) {
-      ;(async () => {
-        // Skipping outbound detection since there's no outbound tx involved here - as long as the inner swap is confirmed, we're gucci
-        const thorchainTxStatus = await waitForThorchainUpdate({
-          txId: confirmedTransaction.txid,
-          skipOutbound: true,
-        }).promise
-
-        if ([TxStatusType.Confirmed, TxStatusType.Failed].includes(thorchainTxStatus)) {
-          contextDispatch({
-            type: ThorchainSaversDepositActionType.SET_DEPOSIT,
-            payload: {
-              txStatus: thorchainTxStatus === TxStatusType.Confirmed ? 'success' : 'failed',
-            },
-          })
+      if ([TxStatusType.Confirmed, TxStatusType.Failed].includes(thorchainTxStatus)) {
+        if (thorchainTxStatus === TxStatusType.Confirmed) {
+          appDispatch(
+            getOpportunitiesUserData.initiate(
+              [
+                {
+                  accountId,
+                  defiProvider: DefiProvider.ThorchainSavers,
+                  defiType: DefiType.Staking,
+                },
+              ],
+              { forceRefetch: true },
+            ),
+          )
```

Comment:
Same here, we want to dispatch this *before* the polling promise resolves, or this will void the race condition fix intent we're trying to do here